sdk/eidolon_ai_sdk/cpu/llm/open_ai_llm_unit.py

```diff
@@ -154,18 +157,16 @@ async def execute_llm(
         tools: List[LLMCallFunction],
         output_format: Union[Literal["str"], Dict[str, Any]],
     ) -> AsyncIterator[AssistantMessage]:
-        if not self.llm:
-            self.llm = AsyncOpenAI()
         yield StartLLMEvent()
-
         try:
             can_stream_message, request = await self._build_request(messages, tools, output_format)
             request["stream"] = True
 
             logger.info("executing open ai llm request", extra=request)
             if logger.isEnabledFor(logging.DEBUG):
                 logger.debug("request content:\n" + yaml.dump(request))
-            llm_response = await self.llm.chat.completions.create(**request)
+            llm_request = replayable(fn=_openai_completion, name_override="openai_completion", parser=_raw_parser)
```

Comment:
This is how we hook up a replay point. Wrap it up with a replayable. There are some types of objects which can't be pickled, which is why we are now instantiating the AsyncOpenAI during the calls.